### PR TITLE
NO-JIRA: resourcebuilder: configv1/infrastructure: allow setting more status fields

### PR DESCRIPTION
**Pull request by damdo**
Allow setting more status fields for the configv1/infrastructure.

### CODE REVIEW

File: testutils/resourcebuilder/config/v1/infrastructure.go [hunk 1]
```diff
@@ -382,3 +393,20 @@ func (i InfrastructureBuilder) WithNamespace(namespace string) InfrastructureBui
...
+	if i.status == nil {
+		i.status = &configv1.InfrastructureStatus{}
+	}
+
+	i.status.PlatformStatus = &ps
+
+	return i
+}
+
+// WithNamespace sets the status for the infrastructure builder.
```
Comment (elmiko) [inline testutils/resourcebuilder/config/v1/infrastructure.go:408]: update the doc string here, also do we need a nil check here as well?

**Review by elmiko — COMMENTED** [review]
minor nit, otherwise lgtm

File: testutils/resourcebuilder/config/v1/infrastructure.go (hunk 1 above)
Comment (damdo) [reply testutils/resourcebuilder/config/v1/infrastructure.go]: Ahh thanks for catching this. Fixed.

**Review by elmiko — COMMENTED** [review]
/lgtm

**Pull request merged**